frame/bounties/src/lib.rs

```diff
@@ -653,12 +786,656 @@ decl_module! {
+		/// else call get failed.
+		///
+		/// Upperbount to maximum number of subbounties that can be added is
+		/// managed via runtime trait config 'MaxSubBountyCount'.
+		///
+		/// Payment: `TipReportDepositBase` will be reserved from the origin account, as well as
+		/// `DataDepositPerByte` for each byte in `reason`. It will be unreserved upon approval,
+		/// or slashed when rejected.
+		///
+		/// if call is success, state of subbounty is moved to "Approved" state.
+		/// And later moved to "Funded" state as part of "spend_fund()" callback.
+		///
+		/// - `bounty_id`: Bounty ID for which subbounty to be added.
+		/// - `value`: Value for executing the proposal.
+		/// - `description`: Text description for the subbounty.
+		#[weight = 10_000]
+		fn add_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			value: BalanceOf<T>,
+			description: Vec<u8>,
+		)
+		{
+			let signer = ensure_signed(origin)?;
+
+			// TODO :: Have to recheck
+			// Governance design choice, which is more appropriate for the context ?
+			//
+			// * add_subbounty() :: right now origin must me signed
+			// & should be master curator. and state of subbounty
+			// is moved to `Approved` state.
+			//
+			// But logic can be refactored, similar to parent bounty sequence flow...
+			//
+			// * propose_subbounty() :: origin may be signed, bond deposit is
+			// applicable & state of subbonty is moved to `Proposed` state.
+			//
+			// * approve_subbounty() :: origin is signed & should be master-curator,
+			// & move the state of subounty to `Approved`.
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let mut bounty = maybe_bounty.as_mut().ok_or(Error::<T>::InvalidIndex)?;
+				match bounty.status {
+					BountyStatus::Active {
+						ref curator,
+						..
+					} => {
+						ensure!(signer == *curator, Error::<T>::RequireCurator);
+						// Verify the arguments
+						ensure!(description.len() <=
+							T::MaximumReasonLength::get() as usize,
+							Error::<T>::ReasonTooBig,
+						);
+						ensure!(value >= T::BountyValueMinimum::get(),
+							Error::<T>::InvalidValue,
+						);
+						ensure!((bounty.activesubbounty.len() as u32) <
+							T::MaxSubBountyCount::get() as u32,
+							Error::<T>::SubBountyMaxOverflow,
+						);
+						// reserve deposit for new subbounty
+						let bond = T::BountyDepositBase::get()
+							+ T::DataDepositPerByte::get() * (description.len() as u32).into();
+						let balance = T::Currency::free_balance(&signer);
+						ensure!(bond < balance, Error::<T>::InsufficientProposersBalance);
+
+						// Makesure Parent bounty have enough balance to fund Subbounty
+						let bounty_account = Self::bounty_account_id(bounty_id);
+						let balance = T::Currency::free_balance(&bounty_account);
+						if_std! {
+							println!("Bounty balance-{:#?}", balance);
+						}
+						ensure!(value <= balance, Error::<T>::InsufficientBountyBalance);
```

Comment:
never check like this, use `ensure_can_withdraw`

Reply:
Thanks, Nice one, will bring into my coding practice.

Reply:
Changes done, ready for review.